components/renku_data_services/message_queue/avro_models/io/renku/events/v1/project_authorization_added.py

```diff
@@ -0,0 +1,64 @@
+from dataclasses import asdict, dataclass
+from typing import ClassVar, Dict
+
+from dataclasses_avroschema import AvroModel
+from renku_data_services.message_queue.avro_models.io.renku.events.v1 import \
+    ProjectMemberRole
+from undictify import type_checked_constructor
+
+
+@type_checked_constructor()
+@dataclass
+class ProjectAuthorizationAdded(AvroModel):
+    """
+    Event raised when an authorization for a project is added for a user
+    """
+    projectId: str
+    userId: str
+    role: ProjectMemberRole
+
+    #: The Avro Schema associated to this class
+    _schema: ClassVar[str] = """{
+        "type": "record",
+        "name": "ProjectAuthorizationAdded",
+        "namespace": "io.renku.events.v1",
+        "doc": "Event raised when an authorization for a project is added for a user",
+        "fields": [
+            {
+                "name": "projectId",
+                "type": "string"
+            },
+            {
+                "name": "userId",
+                "type": "string"
+            },
+            {
+                "name": "role",
+                "type": "io.renku.events.v1.ProjectMemberRole"
+            }
+        ]
+    }"""
+
+    def serialize_json(self) -> str:
```

Comment:
This method and the following two repeat in these classes. Should we make a super class and factor them into that class?

Reply:
everything in message_queue/avro_models is generated code